### ISSUE TITLE
Array configured as camelCased option but value passed as hyphenated flag does't work

### ISSUE THREAD

**Issue by laggingreflex** [yargs/yargs#757]
Test setup:
```js
const config = require('yargs').options({
  sourceFiles: {
    type: 'array'
  }
}).argv

console.log(config.sourceFiles);
```

Correct behavior with "sourceFiles"
```sh
$ node . --sourceFiles 1 2 3
{ _: [], sourceFiles: [ 1, 2, 3 ], '$0': '' }
```

Incorrect behavior with "source-files"
```sh
$ node . --source-files 1 2 3
{ _: [ 2, 3 ],
  'source-files': [ 1 ],
  sourceFiles: [ 1 ],
  '$0': '' }
```

**Comment by laggingreflex** [comment]
Oh, I should've been using hyphenated in config. 
```js
const config = require('yargs').options({
  'source-files': {
    type: 'array'
  }
}).argv
```
Weird, I always used it the other way around. Nevermind

**Comment by bcoe** [comment]
@laggingreflex I would expect both to work honestly, I think this is a bug.

**Comment by laggingreflex** [comment]
This is only an issue with Arrays. 

Proposed a fix in #757

**Comment by bcoe** [comment]
see: https://github.com/yargs/yargs-parser/issues/80
